src/framework/components/button/system.js

```diff
@@ -48,6 +50,15 @@ Object.assign(pc, function () {
             pc.ComponentSystem.prototype.initializeComponentData.call(this, component, data, _schema);
         },
 
+        onUpdate: function (dt) {
+            var components = this.store;
+
+            for (var id in components) {
+                var component = components[id].entity.button;
+                component.onUpdate();
```

Comment:
Would it be a bit faster to do the if statements here instead of calling onUpdate and doing them in there?